src/cleanfigure.m

```diff
@@ -777,9 +777,9 @@ function simplifyLine(meta, handle, targetResolution)
     yData = log10(yData);
   end
 
-  % Turn the data into a row vector
-  xData = xData(:);
-  yData = yData(:);
+  % Limit the precision of the data
+  xData = limitPrecision(xData(:));
+  yData = limitPrecision(yData(:));
```

Comment:
I think I see why nothing is happening: you are limiting the precision on the transformed data, i.e. the visual data, which you don't write back to the objects (or rather, that is what I expect, that you don't write back this transformed data).